### PR TITLE
Add nickname support for `/roster remove`

### PR DESCRIPTION
**Pull request by H3rnand3zzz**
See commit messages.

### CODE REVIEW

**Comment by jubalh** [comment]
Do I see it right that this removed the possibility to remove a user based on JID?

**Comment by H3rnand3zzz** [comment]
> Do I see it right that this removed the possibility to remove a user based on JID?

It in fact did not remove this opportunity. It's covered by tests and I just tested it out manually. It only adds ability to remove user by nickname as well.

File: src/command/cmd_funcs.c [hunk 1]
```diff
@@ -2922,14 +2922,17 @@ cmd_roster(ProfWin* window, const char* const command, gchar** args)
...
+        char* usr = args[1];
+        if (usr == NULL) {
             cons_bad_cmd_usage(command);
-        } else {
-            roster_send_remove(jid);
+            return TRUE;
         }
+        char* barejid = roster_barejid_from_name(usr);
+        if (barejid == NULL) {
+            barejid = usr;
```
Comment (H3rnand3zzz) [inline src/command/cmd_funcs.c:2932]: this lines secures the deletion if name was not found (hence jid was used)

**Review by jubalh — APPROVED** [review]

**Comment by jubalh** [comment]
Thank you!

**Pull request merged**